From 7237bc3d5187b535d402182df5d9da402c7ff215 Mon Sep 17 00:00:00 2001
From: Takeshi Saito <takeshi.saito.xv@renesas.com>
Date: Wed, 20 Mar 2019 18:17:09 +0900
Subject: [PATCH 788/909] arm64: dts: r8a77990-ebisu{,-es10}: Add no-sdio and
 no-sd properties to SDHI3

commit f3f36384af2aad6e7a7279b62297ddf618e8296c from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Add the 'no-sdio' and 'no-sd' properties to the SDHI3 node used as the
eMMC port on Ebisu and Ebisu-4D boards. It will limits the send sdio and
sd cmd during mmc initialization.

In addition, the 'no-sdio' property uses multiple SG entries when SDHI
internal DMAC with IOMMU is enabled. This can improve performance.

Signed-off-by: Takeshi Saito <takeshi.saito.xv@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts      | 2 ++
 arch/arm64/boot/dts/renesas/r8a77990-es10-ebisu.dts | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts b/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
index ef343ffd058b..ae88d6f02115 100644
--- a/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
@@ -797,6 +797,8 @@
 	mmc-hs200-1_8v;
 	mmc-hs400-1_8v;
 	bus-width = <8>;
+	no-sd;
+	no-sdio;
 	non-removable;
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a77990-es10-ebisu.dts b/arch/arm64/boot/dts/renesas/r8a77990-es10-ebisu.dts
index 087d5533f3db..b85df488180a 100644
--- a/arch/arm64/boot/dts/renesas/r8a77990-es10-ebisu.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77990-es10-ebisu.dts
@@ -797,6 +797,8 @@
 	mmc-hs200-1_8v;
 	mmc-hs400-1_8v;
 	bus-width = <8>;
+	no-sd;
+	no-sdio;
 	non-removable;
 	status = "okay";
 };
-- 
2.17.1

